druntime/src/core/time.d

```diff
@@ -1351,6 +1351,11 @@ public:
                 }
 
                 SplitUnits su = void;
+                /* This winds up calling split(Args...), passing it multiple arguments
+                 * that are `out su.unit`, which dip1021 trips as passing multiple mutable
+                 * references. Hence split() is marked @trusted. A better solution would be
+                 * to make multiple calls to split() instead, one for each reference.
+                 */
```

Comment:
This comment is wrong insofar as it would make no sense to make multiple calls to `split`. The entire point is that you're giving it variables which correspond to each of the units, and it splits the time into those respective units. And the way that the time is split depends on which units you instantiate the template with. If the code were changed to make multiple calls to `split`, the result would be different.

```
    auto d = seconds(287122);

    {
        int hours;
        int minutes;
        int seconds;
        d.split!("hours", "minutes", "seconds")(hours, minutes, seconds);
        assert(hours == 79);
        assert(minutes == 45);
        assert(seconds == 22);
    }

    {
        int minutes;
        int seconds;
        d.split!("minutes", "seconds")(minutes, seconds);
        assert(minutes == 4785);
        assert(seconds == 22);
    }
```

Note that the minutes changed in the second call, because `"hours"` was not requested, and the portion of the time that was split out into hours in the first call was then included in the minutes in the second call. Having multiple `out` parameters is fundamental to how `split` works.

In this particular overload of `split`, instead of the caller passing individual variables, a struct is being generated with the requested units, and that struct is returned. So,

```
auto result = d.split!("hours", "minutes", "seconds")();
```

would return a struct that looked like

```
struct SplitUnits
{
    long hours;
    long minutes;
    long seconds;
}
```

and the exact set of member variables would depend on which units you passed to `split` as template arguments.

Before returning the struct, those individual members are passed to the overload of `split` that takes the individual variables so that they can be filled in, e.g.

```
split!("hours", "minutes", "seconds")(su.hours, su.minutes, su.seconds)`;
```

These are individual member variables and are _not_ references to the same data even if they're all member variables of the same struct, so I don't see why that would be triggering DIP 1021. Based on my limited understanding of the DIP, this seems like a bug in the DIP's implementation.

Now, even if it's not a bug in DIP 1021's implementation, slapping `@trusted` on `split` shouldn't be a problem (it probably already should have had it, because of `SplitUnits su = void`), but the comment is wrong regardless.

Reply:
Thanks for the explanation. The compiler implementation could be adjusted to allow non-overlapping refs into the same struct. I'll have to think about it.

Reply:
The problem can be summed up with:
```
@safe:
struct S { int a, b; }
void foo(out int x, ref int y);
void test() {
    S s;
    foo(s.a, s.b);
}
```
```
Error: more than one mutable reference to `s` in arguments to `test3.foo()`
```

Reply:
@jmdavis I fixed the compiler, which now compiles sys.core.time intact.

Reply:
Please put that fix in its own PR